src/community/label_propagation.c

```diff
@@ -213,163 +609,35 @@ igraph_error_t igraph_community_label_propagation(const igraph_t *graph,
-                    }
-                    was_zero = (VECTOR(label_counters)[k] == 0);
-                    VECTOR(label_counters)[k] += VECTOR(*weights)[VECTOR(*ineis)[j]];
-                    if (was_zero && VECTOR(label_counters)[k] != 0) {
-                        /* counter just became nonzero */
-                        IGRAPH_CHECK(igraph_vector_int_push_back(&nonzero_labels, k));
-                    }
-                    if (max_count < VECTOR(label_counters)[k]) {
-                        max_count = VECTOR(label_counters)[k];
-                        IGRAPH_CHECK(igraph_vector_int_resize(&dominant_labels, 1));
-                        VECTOR(dominant_labels)[0] = k;
-                    } else if (max_count == VECTOR(label_counters)[k]) {
-                        IGRAPH_CHECK(igraph_vector_int_push_back(&dominant_labels, k));
-                    }
-                }
-            } else {
-                neis = igraph_adjlist_get(&al, v1);
-                num_neis = igraph_vector_int_size(neis);
-                for (j = 0; j < num_neis; j++) {
-                    k = VECTOR(*membership)[VECTOR(*neis)[j]];
-                    if (k == 0) {
-                        continue;    /* skip if it has no label yet */
-                    }
-                    VECTOR(label_counters)[k]++;
-                    if (VECTOR(label_counters)[k] == 1) {
-                        /* counter just became nonzero */
-                        IGRAPH_CHECK(igraph_vector_int_push_back(&nonzero_labels, k));
-                    }
-                    if (max_count < VECTOR(label_counters)[k]) {
-                        max_count = VECTOR(label_counters)[k];
-                        IGRAPH_CHECK(igraph_vector_int_resize(&dominant_labels, 1));
-                        VECTOR(dominant_labels)[0] = k;
-                    } else if (max_count == VECTOR(label_counters)[k]) {
-                        IGRAPH_CHECK(igraph_vector_int_push_back(&dominant_labels, k));
-                    }
-                }
-            }
-
-            if (igraph_vector_int_size(&dominant_labels) > 0) {
-                if (control_iteration) {
-                    /* Check if the _current_ label of the node is also dominant */
-                    if (VECTOR(label_counters)[VECTOR(*membership)[v1]] != max_count) {
-                        /* Nope, we need at least one more iteration */
-                        running = true;
-                    }
-                }
-                else {
-                    /* Select randomly from the dominant labels */
-                    k = RNG_INTEGER(0, igraph_vector_int_size(&dominant_labels) - 1);
-                    VECTOR(*membership)[v1] = VECTOR(dominant_labels)[k];
-                }
-            }
+      case IGRAPH_LPA_RETENTION:
+        IGRAPH_CHECK(igraph_i_community_label_propagation(graph, membership, mode, weights, fixed_copy, /* retention */ 1 ));
```

Comment:
Please use `true` and `false` instead of `1` and `0` everywhere in this file. This does make a real difference for readability. After having changed so much of the existing code to a more readable form, let's make sure new code uses readable forms that encode the type everywhere.

Same applied to `0` vs `NULL` for pointers.

Reply:
Yes, good point. This was indeed mostly a remnant of old code.